addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/report/AuthReportData.java

```diff
@@ -32,13 +33,46 @@
 import lombok.Setter;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
 import org.zaproxy.addon.authhelper.internal.db.Diagnostic;
 import org.zaproxy.addon.authhelper.internal.db.TableJdo;
 
 @Getter
 @Setter
 public class AuthReportData implements Closeable {
 
+    private static final String KEY_PREFIX = "auth.failure.";
+
+    @Getter
+    public enum FailureDetail {
+        OVERALL(
+                Constant.messages.getString(
+                        "authhelper.authreport.summary.fail.detail.overall.failed")),
```

Comment:
It could do the get in the constructor, it would avoid duplication (the prefix could also be extracted).